presto-product-tests-utils/src/test/java/com/facebook/presto/tests/querystats/TestHttpQueryStatsClient.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.tests.querystats;
+
+import com.facebook.presto.execution.QueryStats;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.io.Resources;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.ResponseHandler;
+import io.airlift.http.client.testing.TestingResponse;
+import io.airlift.json.ObjectMapperProvider;
+import org.assertj.core.api.Assertions;
+import org.mockito.ArgumentCaptor;
```

Comment:
We generally avoid using mockito (or any other mocking framework) because it makes tests brittle and less readable. We leave it as a last resort for testing against libraries that don't have good abstraction (e.g., via interfaces, etc).

Take a look at airlift's TestingHttpClient, which we created for the specific purpose of testing code that relies on airlift's http client.